### DISCUSSION TITLE
Making jingle playlist play as single track

### DISCUSSION THREAD

**Discussion by zachryit** [savonet/liquidsoap#2678] [General]
Please can you assist me with this
`radio = rotate(weights=[1, 4], [jingles, music])
`
Please is it possible to play all tracks in the jingles playlist (instead of 1) then play 4 tracks in the music playlist?
Example 
`all:4 => jingles:music`

**Comment by zachryit** [comment]
Solution by @toots 

Ok. Do you have control over the playlist items? I think a simple way would be to build a source where all the tracks in the playlist are merged into one.

Say, your playlist looks like this:

> annotate:index="1":/path/to/file1.mp3
annotate:index="2":/path/to/file1.mp3
annotate:index="3":/path/to/file1.mp3
annotate:index="4":/path/to/file1.mp3
annotate:index="5":/path/to/file1.mp3


You can use the index to figure out whether or not you should merge tracks using a crossfade:

```
s = playlist("/path/to/playlist/file")

def transition(old, new) =
  s = sequence([old.source, new.source])
 
  old_index = int_of_string(old.metadata["index"])
  new_index = int_of_string(new.metadata["index"])

   #New track is next track: merge!
  if old_index <= new_index then
    merge_tracks(s)
  else
    s
  end
end

s = cross(transition, s)
```

This also assumes that your playlist is set to play tracks in order btw

**Comment by smimram** [comment]
You have access to the number of items in a playlist via the `length` method. You could thus try something like
```
jingles = playlist(...)
music = ...
radio = rotate(weights=[jingles.length(), 4], [jingles, music])
```

**Reply by zachryit** [reply]
Thank you @smimram, it works. This is much simpler. How come I didn't think of this :)

**Reply by toots** [reply]
This works too. The reason I didn't suggest this one at first is b/c this won't work if the playlist is reloaded and the length changes. Actually, thinking about it more, the weights are getters so this should be even better:
```
jingles = playlist(...)
music = ...
radio = rotate(weights=[jingles.length, {4}], [jingles, music])
```

**Reply by zachryit** [reply]
The jingle playlist changes a lot. New jingles are added and old once removed at least every 2 weeks. I probably will have to do more testing in order to conclude. @toots Thanks for the heads-up.

**Comment by wezo3** [comment]
Can I do like this in switch jingle ?
for example
```
jingleL_1 = playlist(loop=false, mode="randomize", reload_mode="watch", "/home/path")
jingle_1 = switch([({ 0h07 }, jingleL_1)])
radio = fallback(track_sensitive=false, [jingle_1])
```
i need to play this playlist until end 
the example I sent it after 1 min its finished auto without resume all list
but I need the jingle playlist start at the time and play all list under it and close or switch after done

if possible can you write example for this ?

**Reply by wezo3** [reply]
@toots can you help please.
